vllm/v1/core/sched/scheduler.py

```diff
@@ -131,14 +132,19 @@ def __init__(
         )
 
         # NOTE(woosuk): Here, "encoder" includes the vision encoder (and
-        # projector if needed). Currently, we assume that the encoder also
-        # has the Transformer architecture (e.g., ViT).
+        # projector if needed) for MM models as well as encoder-decoder
+        # transformers.
         self.max_num_encoder_input_tokens = encoder_compute_budget
         # NOTE: For the models without encoder (e.g., text-only models),
         # the encoder cache will not be initialized because cache size is 0
         # for these models.
-        self.encoder_cache_manager = EncoderCacheManager(
-            cache_size=encoder_cache_size)
+        self.encoder_cache_manager: Optional[EncoderCacheManager] = None
+        if not self.is_encoder_decoder:
+            # An encoder-decoder model does not use the encoder cache.
+            # It uses bidirectional attention and inputs are only
+            # processed once per request.
+            self.encoder_cache_manager = EncoderCacheManager(
+                cache_size=encoder_cache_size)
```

Comment:
I wonder whether a way to keep scheduler cleaner would be passing this `is_encoder_decoder` flag to the cache which makes all methods no-ops and returns `free_encoder_input_ids->[ ]`..

Reply:
Yeah, that sounds nicer. I'll give that a try.

Reply:
done!